src/plugins/mer/merbuildengineoptionswidget.cpp

```diff
@@ -731,8 +731,15 @@ void MerBuildEngineOptionsWidget::onSshTimeoutChanged(int timeout)
 
 void MerBuildEngineOptionsWidget::onSshPortChanged(quint16 port)
 {
+    BuildEngine *const buildEngine = m_buildEngines[m_virtualMachine];
+
     //store keys to be saved on save click
-    m_sshPort[m_buildEngines[m_virtualMachine]] = port;
+    m_sshPort[buildEngine] = port;
+
+    const bool vmOff = (buildEngine->virtualMachine() == nullptr
+                        || buildEngine->virtualMachine()->isOff());
+    m_ui->buildEngineDetailsWidget->setSshPortOccupied(
+                vmOff && Sfdk::isPortOccupied(buildEngine->sshPort()));
```

Comment:
I believe `buildEngine->virtualMachine()` is guaranteed to be always non-null.

I think the condition is not fully correct, maybe:

```suggestion
    m_ui->buildEngineDetailsWidget->setSshPortOccupied(
                (buildEngine->virtualMachine()->isOff() || port != buildEngine->sshPort())
                && Sfdk::isPortOccupied(port));
```

Reply:
Ok, your variant is better. Fixed.